### PR TITLE
[Refactored] : Update cses-1085.mdx

### PR DESCRIPTION
**Pull request by amansharma07**
Modified search space for binary search as left can be maximum element in array and right can be the sum of all elements in array.

_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
honestly i'd argue that this is a bit more convoluted but it might just be me

**Comment by amansharma07** [comment]
Well we should try to reduce the search space but yeah I agree this doesn't improve the time or anything. Just makes it faster on those edge cases where values are in a very small range and count is high.

**Comment by TheGamingMousse** [comment]
> Well we should try to reduce the search space but yeah I agree this doesn't improve the time or anything. Just makes it faster on those edge cases where values are in a very small range and count is high.

for binary search it practically never matters tho

but I'm fine with this

**Comment by SansPapyrus683** [comment]
wait since there's java & python versions i think it's best that we keep the constant

**Review by ryanchou-dev — COMMENTED** [review]
if you can make the changes for java/python too, that'd be great 👀

**Comment by SansPapyrus683** [comment]
can i have proof of ac of these sols

**Comment by amansharma07** [comment]
Proof of AC attached.

CSES-Array-Division-Results (2)
CSES-Array-Division-Results (1)
CSES-Array-Division-Results

File: solutions/silver/cses-1085.mdx [hunk 1]
```diff
@@ -71,9 +68,14 @@ int main() {
 	int n, k;
 	cin >> n >> k;
 	vector<ll> arr(n);
-	for (ll &i : arr) { cin >> i; }
+	ll sum = 0, mx = 0;
```
Comment (SansPapyrus683) [inline solutions/silver/cses-1085.mdx:71]: i'd rather you use `std::max_element` and `std::accumulate` instead

Comment (amansharma07) [reply solutions/silver/cses-1085.mdx]: Updated now @SansPapyrus683

**Review by amansharma07 — COMMENTED** [review]
Changed

File: solutions/silver/cses-1085.mdx [hunk 2]
```diff
@@ -73,7 +70,8 @@ int main() {
 	vector<ll> arr(n);
 	for (ll &i : arr) { cin >> i; }
 
-	ll l = 1, r = MAX_SUM;
+	ll l = *max_element(begin(arr), end(arr)),
+	   r = accumulate(begin(arr), end(arr), 0LL);
```
Comment (TheGamingMousse) [inline solutions/silver/cses-1085.mdx:74]: why not just have 

```cpp
ll l = *max_element(begin(arr), end(arr));
ll r = accumulate(begin(arr), end(arr), 0LL);
```

**Review by TheGamingMousse — APPROVED** [review]
looks good to me

**Comment by SansPapyrus683** [comment]
send proof of ac again

**Comment by amansharma07** [comment]
AC Proofs

C++ : 
image

Java :
image

Python :
image

**Pull request merged**